diff --git a/core/jni/android_server_BluetoothEventLoop.cpp b/core/jni/android_server_BluetoothEventLoop.cpp
index aa981fe..6f093dc 100644
--- a/core/jni/android_server_BluetoothEventLoop.cpp
+++ b/core/jni/android_server_BluetoothEventLoop.cpp
@@ -341,7 +341,7 @@ const char * get_adapter_path(DBusConnection *conn) {
                     "org.freedesktop.DBus.Error.ServiceUnknown")) {
                     // bluetoothd is still down, retry
                     LOG_AND_FREE_DBUS_ERROR(&err);
-                    usleep(10000);  // 10 ms
+                    usleep(100000);  // 100 ms
                     continue;
                 } else {
                     // Some other error we weren't expecting
