diff --git a/libalsa-intf/alsa_ucm.c b/libalsa-intf/alsa_ucm.c
index a312470..9d1e132 100644
--- a/libalsa-intf/alsa_ucm.c
+++ b/libalsa-intf/alsa_ucm.c
@@ -702,6 +702,12 @@ int use_case_index)
                     tx_id = DEVICE_SPEAKER_TX_FV5_ACDB_ID;
                 }
 
+                if (rx_id == DEVICE_HANDSET_RX_ACDB_ID &&
+                           tx_id == DEVICE_SPEAKER_TX_ACDB_ID) {
+                    tx_id = DEVICE_HANDSET_TX_ACDB_ID;
+                    ALOGD("Varcain: mic workaround");
+                }
+
                 if ((rx_id != uc_mgr->current_rx_device) ||
                     (tx_id != uc_mgr->current_tx_device)) {
                     uc_mgr->current_rx_device = rx_id;
